---
title: "Introduction to genomic data"
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
 encoding=encoding,
 output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
author: "Todd W. Pierson"
date: "8/28/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this tutorial, we'll work through a basic introduction to handling raw genomic data. This will include brief lessons on:

1) formatting of FASTQ files;
2) multiplexing strategies and how to demultiplex reads; and
3) quality-filtering


## Interpreting a FASTQ file

Most raw genomic data come in the FASTQ format. For comparison, it is worth reviewing the structure of a FASTA format. 

We can peek at the first two lines of an example FASTA file of the data directory.
```{bash}
head -n 2 data/example.fasta
```

Now, for FASTQ files.
```{bash}
head -n 4 data/example.fastq
```

## Multiplexing and demultiplexing

[ placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here ]

## Quality filtering

[ placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here placeholder text here ]
